client/src/hooks/useStakeholders/useStakeholders.js

```diff
@@ -47,77 +49,94 @@ export function useStakeholders() {
-            selectedCategories,
-            selectedDistance
-          );
-        }
-      } catch (error) {
-        dispatch({ type: FETCH_FAILURE, error });
-      }
-    };
+  const fetchLocation = async () => {
+    const {
+      FETCH_FAILURE,
+      FETCH_REQUEST,
+      FETCH_SUCCESS
+    } = actionTypes.LOCATION;
 
-    // The fact that the geolocation api only uses callbacks makes
-    // this function convoluted - need to work on cleaning this up
-    // somehow. If user agent (browser) does not support geolocation
-    // of machine, or user disables location, we currently use the
-    // lat, lon of LACI as the current location. Need to work on
-    // allowing user to enter a street address of their choosing.
-    if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition(
-        async position => {
-          if (position) {
-            let latitude = position.coords.latitude;
-            let longitude = position.coords.longitude;
-            fetchCategories(latitude, longitude);
+    dispatch({ type: FETCH_REQUEST });
+    try {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(
+          async position => {
+            if (position) {
+              const userCoordinates = {
+                latitude: position.coords.latitude,
+                longitude: position.coords.longitude
+              };
+              dispatch({ type: FETCH_SUCCESS, userCoordinates });
+            }
+            // async fn must return something
+            return { latitude: null, longitude: null };
+          },
+          async error => {
+            dispatch({ type: FETCH_FAILURE, error });
+            return { latitude: null, longitude: null };
           }
-        },
-        async error => {
-          console.log(error);
-          fetchCategories();
-        }
-      );
-    } else {
-      fetchCategories();
+        );
+      } else {
+        // If browser location permission is denied, the request is
+        // "successful", but the result is null coordinates.
+        dispatch({
+          type: FETCH_SUCCESS,
+          userCoordinates: { latitude: null, longitude: null }
+        });
+      }
+    } catch (error) {
+      dispatch({ type: FETCH_FAILURE, error });
+      return error;
     }
+  };
+
+  useEffect(() => {
+    const {
+      searchString,
+      selectedLatitude,
+      selectedLongitude,
+      selectedLocationName,
+      selectedDistance,
+      selectedCategories
+    } = initialState;
+
+    // Runs once on initialization to get list of all active categories
+    fetchCategories();
+
+    // Runs once on initialization to get user's browser lat/lon, if
+    // browser permits
+    fetchLocation();
+
+    // Exposed to consuming component to execute search
+    search(
+      searchString,
+      selectedLatitude,
+      selectedLongitude,
+      selectedLocationName,
+      selectedCategories,
+      selectedDistance
+    );
```

Comment:
this part is interesting... when I pulled down and ran locally, it went ahead and searched based on the initialState lat/long of the LACI coordinates, even though I've permitted the browser to get my location. the `search(...)` is always using the initialState values. this was the tricky part when i initially wrote the custom hook because you want to validate the lat/long, be they initialState (aka LACI) or the verified user location. perhaps it's somewhat moot once we have the geocoder autocomplete. when i did some research on other popular map/search/list-results apps (google maps, yelp, etc), usually they just load up a somewhat 'empty' map without grabbing your location at all, and then you go from there.

Reply:
ok I think I see what you're doing here. really just fetching the location so you can have that as a radio button option in the filters, yeah?